packages/forms/src/UIForm/fieldsets/Array/Array.component.js

```diff
@@ -127,7 +127,7 @@ export default class ArrayWidget extends React.Component {
 
 	render() {
 		const { schema } = this.props;
-		const canReorder = schema.reorder !== false;
+		const canReorder = !schema.readOnly && schema.reorder !== false;
```

Comment:
Disabled is managed in ArrayItem.component.js, I think we can manage all those actions in the same file too.
Moreover you already add the delete button if it's not in readonly mode.